File: backend/src/schemas/chart.py
from typing import Dict, Any, List, Optional
from datetime import datetime
from pydantic import BaseModel, Field


class ChartBase(BaseModel):
    """图表基础模式"""
    title: str = Field(..., description="图表标题")
    description: Optional[str] = Field(None, description="图表描述")
    type: str = Field(..., description="图表类型")
    config: Dict[str, Any] = Field(..., description="图表配置")
    data_source: Optional[Dict[str, Any]] = Field(None, description="数据源配置")


class ChartCreate(ChartBase):
    """创建图表模式"""
    analysis_task_id: Optional[str] = Field(None, description="分析任务ID")


class ChartUpdate(BaseModel):
    """更新图表模式"""
    title: Optional[str] = Field(None, description="图表标题")
    description: Optional[str] = Field(None, description="图表描述")
    type: Optional[str] = Field(None, description="图表类型")
    config: Optional[Dict[str, Any]] = Field(None, description="图表配置")
    data_source: Optional[Dict[str, Any]] = Field(None, description="数据源配置")


class ChartResponse(ChartBase):
    """图表响应模式"""
    id: str = Field(..., description="图表ID")
    analysis_task_id: Optional[str] = Field(None, description="分析任务ID")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")
    
    class Config:
        orm_mode = True


class ChartTypeResponse(BaseModel):
    """图表类型响应模式"""
    type: str = Field(..., description="图表类型")
    name: str = Field(..., description="图表名称")


class ChartOptionsResponse(BaseModel):
    """图表选项响应模式"""
    options: Dict[str, Any] = Field(..., description="图表选项") 

File: frontend/src/api/chart.ts
import { get, post, put, del } from './index'

// 图表类型定义
export interface Chart {
  id: string
  title: string
  description?: string
  type: string
  config: any
  data_source?: any
  analysis_task_id?: string
  created_at: string
  updated_at: string
}

// 图表类型
export interface ChartType {
  type: string
  name: string
}

// 图表选项
export interface ChartOptions {
  options: any
}

// 创建图表请求
export interface ChartCreateRequest {
  title: string
  description?: string
  type: string
  config: any
  data_source?: any
  analysis_task_id?: string
}

// 更新图表请求
export interface ChartUpdateRequest {
  title?: string
  description?: string
  type?: string
  config?: any
  data_source?: any
}

// 获取图表列表
export function getCharts() {
  return get<Chart[]>('/v1/charts')
}

// 获取单个图表
export function getChart(id: string) {
  return get<Chart>(`/v1/charts/${id}`)
}

// 创建图表
export function createChart(data: ChartCreateRequest) {
  return post<Chart>('/v1/charts', data)
}

// 更新图表
export function updateChart(id: string, data: ChartUpdateRequest) {
  return put<Chart>(`/v1/charts/${id}`, data)
}

// 删除图表
export function deleteChart(id: string) {
  return del<void>(`/v1/charts/${id}`)
}

// 获取图表类型列表
export function getChartTypes() {
  return get<ChartType[]>('/v1/charts/types')
}

// 获取图表选项
export function getChartOptions(type: string) {
  return get<ChartOptions>(`/v1/charts/options/${type}`)
}

// 预览图表
export function previewChart(data: ChartCreateRequest) {
  return post<any>('/v1/charts/preview', data)
}

// 获取图表数据
export function getChartData(id: string) {
  return get<any>(`/v1/charts/${id}/data`)
} 